cypress/integration/InstancesTest.ts

```diff
@@ -79,177 +85,256 @@ describe("Instances Test", () => {
+      cy.get("[aria-label='Go to next page']").should("not.be.disabled");
+      // Go to next page
+      cy.ouiaId("OUIA-Generated-Button-plain-3", "PF4/Button").click();
+      cy.get("[aria-label='Go to first page']").should("not.be.disabled");
+      cy.get("[aria-label='Go to previous page']").should("not.be.disabled");
+      cy.get("[aria-label='Go to last page']").should("be.disabled");
+      cy.get("[aria-label='Go to next page']").should("be.disabled");
+      // Go to previous page
+      cy.ouiaId("OUIA-Generated-Button-plain-2", "PF4/Button").click();
 
-    it("Accepted", () => {
-      const instanceName: string = "Instance three";
-      cy.ouiaId("Instances list table", "PF4/Table")
-        .ouiaId(instanceName, "PF4/TableRow")
-        .should("be.visible")
-        .within(() => {
-          cy.get("td:first").should("have.html", instanceName);
-          cy.get("td:nth-child(2)").should("have.text", "accepted");
-          cy.get("td:nth-child(4)")
-            .click()
-            .within(() => {
-              cy.ouiaType("PF4/DropdownItem").then(($items) => {
-                expect($items.eq(0)).have.text("Details");
-                expect($items.eq(0)).be.enabled;
-                expect($items.eq(1)).have.text("Delete");
-                expect($items.eq(1)).have.attr("aria-disabled", "true");
-              });
-            });
+      cy.ouiaType("PF4/DropdownToggle").should("be.visible");
+      cy.ouiaId(
+        "OUIA-Generated-DropdownToggle-1",
+        "PF4/DropdownToggle"
+      ).click();
+      cy.get(".pf-c-options-menu__menu")
+        .find("li")
+        .then(($cells) => {
+          expect($cells).have.length(4);
+          expect($cells.eq(0)).have.text("10 per page");
+          expect($cells.eq(1)).have.text("20 per page");
+          expect($cells.eq(2)).have.text("50 per page");
+          expect($cells.eq(3)).have.text("100 per page");
         });
+
+      // page size is changed
+      cy.contains("20 per page").click(); // change page-size
+      cy.get(".pf-c-pagination__total-items >b:nth-of-type(1)").then(function (
+        count
+      ) {
+        let size = count.text();
+        let perPageSize = parseInt(size.slice(-2));
+        cy.get(".pf-c-pagination__total-items >b:nth-of-type(2)").then(
+          function (count) {
+            let pageSize = parseInt(count.text());
+            cy.wrap(perPageSize).should("be.lte", pageSize);
+          }
+        );
+      });
+
+      cy.get(".pf-c-pagination__total-items >b:nth-of-type(2)").then(function (
+        count
+      ) {
+        let initialInstanceCount = parseInt(count.text());
+        createInstance();
+        cy.get(".pf-c-pagination__total-items >b:nth-of-type(2)").then(
+          function (count) {
+            let instanceCountAfterCreate = parseInt(count.text());
+            expect(instanceCountAfterCreate).to.be.greaterThan(
+              initialInstanceCount
+            );
```

Comment:
> I am not sure if we could use this approach. It is the first step to create something like Page Fragments which is not recommended by [Cypress - best practice](https://docs.cypress.io/guides/references/best-practices#Organizing-Tests-Logging-In-Controlling-State). The original websites shows plenty of examples how to deal with it.
> 
> I agree that some isolated functions are good to share. IMHO ... delete and create Instance are such functions (two occurences). I think that we can use commands or actions or create separate file (something like Utils.ts). I am not sure if export function from a test file is a good idea.
> 
> Such functions should be:
> 
> * parameterized (name, state, etc.). See my comments.
> * assert the neccesary objects (does we always assert that the instance is in the appropriate state or wait that the state is changed or number of columns)
> 
> Probably large projects are using PageFragments because you can find blog articles like [Stop using Page Objects and Start using App Actions](https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/#page-objects-in-cypress).



> d

Thanks @jomarko and @sjamboro for the guidance and suggestions :)

Worked on the suggested changes. I preferred creating "Utility file- Util.ts" for common functions ( create and delete instance)
and kept these function as generic one  - a parameterized function with asserting necessary objects only

and hence created different function (newInstanceStatus and deleteInstanceStatus) which basically acts as wait of their expected behaviour (like createInstance should be in ready state and deleteInstance should not exist)

Kindly let me know your opinion on this.